components/script/dom/blob.rs

```diff
@@ -88,22 +90,76 @@ impl DataSlice {
 pub struct Blob {
     reflector_: Reflector,
     #[ignore_heap_size_of = "No clear owner"]
-    data: DataSlice,
+    blob_impl: BlobImpl,
     typeString: String,
     isClosed_: Cell<bool>,
 }
 
+#[derive(Clone, JSTraceable)]
+pub struct BlobImpl {
+    slice: DOMRefCell<Option<DataSlice>>,
```

Comment:
I am thinking about the possibility of caching here. If it is file-backed, we can put the cached content inside the other `DataSlice` field.